R/server.R

```diff
@@ -22,9 +22,14 @@ server <- function(elts, opt, custom_admin_panel) {
     if (opt$enable_admin_panel)
       shiny::outputOptions(output, "admin_panel.ui", suspendWhenHidden = FALSE)
 
-    if(!is.null(opt$on_session_ended_fun))
-      session$onSessionEnded(function() {
-        opt$on_session_ended_fun(state)
+    if (!is.null(opt$on_session_ended_fun))
+      session$onSessionEnded(function(...) {
+        opt$on_session_ended_fun(state, ...)
+      })
+
+    if (!is.null(opt$on_stop_fun))
+      session$onStop(function(...) {
```

Comment:
This is not quite right, I recommend rereading the documentation here to clarify the API: https://shiny.rstudio.com/reference/shiny/latest/onStop.html

I'm surprised this didn't throw an error when you tested it?